src/components/Oauth/OAuthWindow/OAuthWindow.tsx

```diff
@@ -0,0 +1,79 @@
+import { useEffect, useState } from 'react';
+
+import { useApiKey } from '../../../context/ApiKeyContextProvider';
+import { useConnections } from '../../../context/ConnectionsContextProvider';
+import { useProject } from '../../../context/ProjectContextProvider';
+
+import { useOAuthWindowToast } from './useOAuthWindowToast';
+import { openWindow, receiveMessageEvent, refreshConnection } from './windowHelpers';
+
+type OAuthWindowProps = {
+  children: React.ReactNode;
+  windowTitle: string;
+  oauthUrl: string | null;
+  onClose: (err: string | null) => void;
+};
+
+/**
+ * v2 OAuth window to manage OAuth flow
+ * @param param0
+ * @returns
+ */
+export function OAuthWindow({
+  children, oauthUrl, windowTitle = 'OAuthWindow', onClose,
+}: OAuthWindowProps) {
+  const apiKey = useApiKey();
+  const { projectId } = useProject();
+  const [connectionId, setConnectionId] = useState(null);
+  const [oauthWindow, setOauthWindow] = useState<Window | null>(null);
+  const { setSelectedConnection } = useConnections();
+  const { ToastSuccessConnection, ToastOauthFailed } = useOAuthWindowToast();
+
+  const receiveMessage = receiveMessageEvent(setConnectionId);
+  const openOAuthWindow = openWindow(windowTitle, setOauthWindow, receiveMessage, oauthUrl);
+  const refreshConnections = refreshConnection(projectId, apiKey, setSelectedConnection);
+
+  // open the OAuth window on mount and prop change
+  useEffect(() => {
+    if (oauthUrl && !oauthWindow) {
+      openOAuthWindow(); // creates new window and adds event listener
+    }
+  }, [oauthUrl, oauthWindow, openOAuthWindow, receiveMessage, windowTitle]);
+
+  // refresh connections on connectionId change
+  useEffect(() => {
+    if (connectionId) {
+      refreshConnections(connectionId)
+        .then(() => {
+          oauthWindow?.close(); // only close the window if connection is successful
+          // show success toast if connection is successful
+          ToastSuccessConnection();
```

Comment:
We shouldn't do a toast, that is too intrusive. Toasts are ok to do in our own console, but not in the builder's app since we have no idea what their layout is, and we should keep our UI within the embeddable component itself & the oauth popup. So please remove `ToastSuccessConnection`, `ToastOauthFailed` and all the other toast related code.